### PR TITLE
Support OBJECT data type in tablet write

### PR DESCRIPTION
**Pull request by shuwenwei**
## Description

Adds support for the **OBJECT** data type (TSDataType `12`) in the table-model tablet write path of the Go client, following the Java `TableSession` / tsfile implementation.

### Write path
- Add `OBJECT` to the `TSDataType` enum and the string/byte type maps (`client/protocol.go`)
- Treat OBJECT as a binary column (same as BLOB) in `Tablet` (`client/tablet.go`): value storage, `SetValueAt`/`GetValueAt`, `Swap`, `getValuesBytes`, `NewTablet`
- Add `Tablet.SetObjectValueAt(isEOF, offset, content, columnIndex, rowIndex)` for **segmented OBJECT writes**: each segment is wrapped with a 1-byte `isEOF` flag + 8-byte big-endian offset, matching Java `Tablet.addValue(rowIndex, columnIndex, isEOF, offset, content)`

### Read path
- Decode OBJECT (and BLOB/STRING) binary columns in `BinaryArrayColumnDecoder`, and handle OBJECT like BLOB in `IoTDBRpcDataSet` getters, so written objects can be read back (e.g. `select READ_OBJECT(file)` + `GetBlob`)

### Tests
- Unit tests: `TestTablet_OBJECT` (whole-object write), `TestTablet_SetObjectValueAt` (segmented write) in `client/tablet_test.go`
- E2E table test: `Test_InsertObjectTablet` in `test/e2e/e2e_table_test.go`, covering whole-object, segmented (512B), and null-object rows, verified via `READ_OBJECT`/`GetBlob`, `count(*)`, null check, and tag/field round-trip

Also includes the regenerated thrift `common` code (new aggregation types, `pipeRecentFailureList` field) produced by `make all`.

## Verification
- `go build ./...` / `go test ./client/...` pass
- E2E suite requires Docker (`make e2e_test`)

### CODE REVIEW

File: client/rpcdataset.go [hunk 1]
```diff
@@ -542,7 +542,7 @@ func (s *IoTDBRpcDataSet) getObjectByTsBlockIndex(tsBlockColumnIndex int32) (int
 		} else {
 			return binary.GetStringValue(), nil
 		}
-	case BLOB:
+	case BLOB, OBJECT:
```
Comment (JackieTien97) [inline client/rpcdataset.go]: [P1] Decode OBJECT results with OBJECT semantics

Grouping `OBJECT` with `BLOB` exposes server-side OBJECT metadata instead of the public object representation. In a live TimechoDB test, `SELECT file` returned an OBJECT payload whose first eight bytes are the object size and whose remaining bytes contain the internal object path; this branch makes `GetObject` return those raw bytes, while the matching `GetString` branch hex-encodes them. The Java client uses `BytesUtils.parseObjectByteArrayToString(...)` for both getters (for example, `(Object) 1.00 KB`) and rejects `getBlob` for OBJECT. Please split OBJECT from BLOB, implement the same formatter/getter contract, add direct `SELECT file` tests, and add the corresponding `database/sql` column mapping so a non-null OBJECT is not silently returned as `nil`.

File: client/tablet.go [hunk 2]
```diff
@@ -195,7 +195,7 @@ func (t *Tablet) SetValueAt(value interface{}, columnIndex, rowIndex int) error
 		default:
 			return fmt.Errorf("illegal argument value %v %v", value, reflect.TypeOf(value))
 		}
-	case TEXT, STRING:
+	case TEXT, STRING, OBJECT:
```
Comment (JackieTien97) [inline client/tablet.go]: [P1] Do not accept unframed OBJECT payloads here

For TableSession OBJECT writes, the server interprets every value as a segment envelope: one EOF byte, an eight-byte big-endian offset, then content. This branch accepts an arbitrary `string` or `[]byte` and serializes it without that envelope; a live insertion of a string accepted here failed with status 741 because the server interpreted the string bytes as the offset. Please remove OBJECT from this generic branch and have `SetObjectValueAt` assign through an internal framed-value helper, or expose a separate, clearly documented API if another valid pre-encoded/object-path representation is required. The unit test should not assert that raw `"hello"` is a valid TableSession OBJECT value.

File: client/tablet.go [hunk 3]
```diff
@@ -229,6 +229,42 @@ func (t *Tablet) SetValueAt(value interface{}, columnIndex, rowIndex int) error
...
+	if t.measurementSchemas[columnIndex].DataType != OBJECT {
+		return fmt.Errorf("column %d must be of type OBJECT", columnIndex)
+	}
+	value := make([]byte, len(content)+9)
+	if isEOF {
+		value[0] = 1
+	}
+	binary.BigEndian.PutUint64(value[1:9], uint64(offset))
+	copy(value[9:], content)
+	return t.SetValueAt(value, columnIndex, rowIndex)
```
Comment (JackieTien97) [inline client/tablet.go]: [P2] Clear a previously marked NULL bit before storing the segment

`SetValueAt(nil, ...)` marks this cell in the bitmap, but the non-nil path never unmarks it. Consequently, calling `SetValueAt(nil, ...)` and then `SetObjectValueAt(...)` for the same cell still writes a NULL row; this was reproducible against TimechoDB. The equivalent Java overload calls `updateBitMap(rowIndex, columnIndex, false)`. Please clear the bit for non-nil assignments (ideally in the shared `SetValueAt` path) and add a regression test for overwriting NULL with an OBJECT value.

File: common/common.go [hunk 4]
```diff
@@ -475,6 +475,10 @@ const (
 	TAggregationType_SKEWNESS TAggregationType = 38
 	TAggregationType_KURTOSIS TAggregationType = 39
 	TAggregationType_PERCENTILE TAggregationType = 40
+	TAggregationType_RATE TAggregationType = 41
```
Comment (JackieTien97) [inline common/common.go:478]: [P2] Keep unrelated generated protocol updates out of this OBJECT PR

These aggregation constants and the `pipeRecentFailureList` additions are generated from a newer IoTDB protocol and are unrelated to OBJECT tablet support. Bundling them expands the public Thrift/protocol surface, creates version drift, and also brings in the correctness issue already reported in the `Equals` implementation below. Please revert `common/common.go` from this PR and submit any pinned, reviewed IDL regeneration as a separate change.

**Review by JackieTien97 — CHANGES_REQUESTED** [review]
The segmented OBJECT write envelope matches the Java implementation and passed live round-trip testing against TimechoDB. However, the public OBJECT read/write semantics and null-bitmap handling still have reproducible correctness issues, and the unrelated generated Thrift changes should be removed from this PR. Please address the inline comments before merging.
